diff --git a/dist/stringify.js b/dist/stringify.js
index 8e3062b89ce2a16c8885ef4a48705113dfc97328..faf11ce6fac1aecd5fd204b3cef472be430c8d67 100644
--- a/dist/stringify.js
+++ b/dist/stringify.js
@@ -60,7 +60,7 @@ const BSON_TO_JS_STRING = {
         }
         // The `Binary.buffer.toString` type says it doesn't accept
         // arguments. However it does, and a test will fail without it.
-        return `BinData(${subType.toString(16)}, '${v.toString('base64')}')`;
+        return `BinData(${subType.toString(10)}, '${v.toString('base64')}')`;
     },
     DBRef: function (v) {
         if (v.db) {
