diff --git a/lib/stringify.js b/lib/stringify.js
index fdecdcd3f741298a22b240dcdffe792f49b1dee3..351cdbd6495bf225acfd2b09650f534ac3972b26 100644
--- a/lib/stringify.js
+++ b/lib/stringify.js
@@ -49,7 +49,8 @@ const BSON_TO_JS_STRING = {
   Binary: function(v) {
     const subType = v.sub_type;
     if (subType === 4) {
-      return `UUID("${v.buffer.toString('hex')}")`;
+      const uuidHex = v.buffer.toString('hex');
+      return `UUID("${uuidHex.slice(0, 8)}-${uuidHex.slice(8, 12)}-${uuidHex.slice(12, 16)}-${uuidHex.slice(16, 20)}-${uuidHex.slice(20, 32)}")`;
     }
     return `BinData(${subType.toString(16)}, '${v.buffer.toString('base64')}')`;
   },
