diff --git a/lib/stringify.js b/lib/stringify.js
index 93904b0d40f23780be14ed3708a1b5f1285e254c..fbdc7dc738f7f3bb55e176ddd2a321cb09e37a47 100644
--- a/lib/stringify.js
+++ b/lib/stringify.js
@@ -49,7 +49,8 @@ const BSON_TO_JS_STRING = {
   Binary: function(v) {
     const subType = v.sub_type;
     if (subType === 4) {
-      return `UUID("${v.buffer.toString('hex')}")`;
+      const uuidHex = v.buffer.toString('hex')
+      return `UUID("${uuidHex.slice(0, 8)}-${uuidHex.slice(8, 12)}-${uuidHex.slice(12, 16)}-${uuidHex.slice(16, 20)}-${uuidHex.slice(20, 32)}")`;
     }
     return `BinData(${subType.toString(16)}, '${v.buffer.toString('base64')}')`;
   },
