diff --git a/node_modules/mongodb-query-parser/lib/index.js b/node_modules/mongodb-query-parser/lib/index.js
index 451a0bb..a550779 100644
--- a/node_modules/mongodb-query-parser/lib/index.js
+++ b/node_modules/mongodb-query-parser/lib/index.js
@@ -41,10 +41,12 @@ const BSON_TO_JS_STRING = {
     return `BinData(${subType.toString(16)}, '${v.buffer.toString('base64')}')`;
   },
   DBRef: function(v) {
-    return `DBRef('${v.namespace}', '${v.oid}')`;
+    // add db as a param
+    return `DBRef('${v.namespace}', '${v.oid}'${typeof v.db === 'string' ? `, '${v.db}'` : ''})`;
   },
   Timestamp: function(v) {
-    return `Timestamp(${v.low}, ${v.high})`;
+    // low_ and high_ matches bson 1.X, which is used by this package
+    return `Timestamp(${v.low_}, ${v.high_})`;
   },
   Long: function(v) {
     return `NumberLong(${v.toString()})`;
